agent/agent.go

```diff
@@ -241,15 +241,43 @@ func (a *Agent) Start() error {
 		return fmt.Errorf("Failed to setup node ID: %v", err)
 	}
 
+	// create the config for the rpc server/client
+	consulCfg, err := a.consulConfig()
+	if err != nil {
+		return err
+	}
+
 	// Setup either the client or the server.
 	if c.Server {
-		server, err := a.makeServer()
+		serverUp := make(chan int)
+		consulCfg.ServerUp = func() { close(serverUp) }
+
+		server, err := consul.NewServerLogger(consulCfg, a.logger)
 		if err != nil {
-			return err
+			return fmt.Errorf("Failed to start Consul server: %v", err)
 		}
 
 		a.delegate = server
-		a.state.Init(c, a.logger, server)
+
+		// wait for the server to be ready
+		// then trigger an anti-entropy run ???
+
+		// this needs review since there is no for the state to be
```

Comment:
ok, the race is in the way the coupling is done since the state is created since I'm now calling `state.Init` after the server has been created. The `consulCh` the server go routine is trying to use doesn't exist at that point. Since the state needs the server and the server needs the state there is some coupling that needs to be guarded.